store/src/java/com/zimbra/cs/service/formatter/LdifFormatter.java

```diff
@@ -0,0 +1,125 @@
+            HashSet<String> fields = new HashSet<String>();
+            UserServletUtil.populateContactFields(contacts, context.targetMailbox, context.opContext, allContacts, fields);
+            ArrayList<String> allFields = new ArrayList<String>();
+            allFields.addAll(fields);
+            Collections.sort(allFields);
+            for (Map<String, String> contactMap : allContacts) {
+                toLDIFContact(allFields, contactMap, sb);
+                sb.append(NEW_LINE);
+            }
+        } finally {
+            if (contacts instanceof QueryResultIterator)
+                ((QueryResultIterator) contacts).finished();
+        }
+        String filename = context.itemPath;
+        if (filename == null || filename.length() == 0)
+            filename = "contacts";
+        if (filename.toLowerCase().endsWith(".ldif") == false) {
+            filename = filename + ".ldif";
+        }
+        String cd = HttpUtil.createContentDisposition(context.req, Part.ATTACHMENT, filename);
+        context.resp.addHeader("Content-Disposition", cd);
+        context.resp.setCharacterEncoding(context.getCharset().name());
+        context.resp.setContentType(MimeConstants.CT_TEXT_LDIF);
+        context.resp.getWriter().print(sb.toString());
+    }
+
+    private void toLDIFContact(List<String> fields, Map<String, String> contact, StringBuilder sb) {
+        for (String f : fields) {
+            String value = contact.get(f);
+            if (value != null) {
+                sb.append(f);
+                sb.append(": ");
+                sb.append(value);
+                sb.append(NEW_LINE);
+            }
+        }
+    }
+
+    /**
+     * Disable the Jetty timeout for the SelectChannelConnector and the
+     * SSLSelectChannelConnector for this request.
+     *
+     * By default (and our normal configuration) Jetty has a 30 second idle
+     * timeout (10 if the server is busy) for connection endpoints. There's
+     * another task that keeps track of what connections have timeouts and
+     * periodically works over a queue and closes endpoints that have been timed
+     * out. This plays havoc with downloads to slow connections and whenever we
+     * have a long pause while working to create an archive.
+     *
+     * This method instructs Jetty not to close the connection when the idle
+     * time is reached. Given that we don't send a content-length down to the
+     * browser for archive responses, we have to close the socket to tell the
+     * browser its done. Since we have to do that.. leaving this endpoint
+     * without a timeout is safe. If the connection was being reused (ie
+     * keep-alive) this could have issues, but its not in this case.
+     * 
+     * @throws IOException
+     */
+    private void disableJettyTimeout(UserServletContext context) {
+        if (LC.zimbra_csv_formatter_disable_timeout.booleanValue()) {
```

Comment:
I would certainly hope that disabling the timeout is unnecessary.  With reasonable limits on the number of allowed entries (zimbraContactMaxNumEntries) and a relatively efficient conversion process that would seem like overkill for most use cases.

Potential RFE: It might be nice to have the ability to disable the timeout only for admin users (and only if they explicitly ask for that), but that perhaps complicates logic too much?